File: Tree_Inorder_Traversal/Tree_Inorder_Traversal.c
#include <stdio.h>
#include <stdlib.h>

typedef struct node
{
	int data;
	struct node *lptr,*rptr;
}node;

node *newnode(int data)
{
	node *newn;
	newn = (node*)malloc(sizeof(node));
	newn->data = data;
	newn->lptr = NULL;
	newn->rptr = NULL;
	return newn;
}

/*Function for INORDER Transversal*/
void inorder(node *root)
{
	if(root != NULL)
	{
		inorder(root->lptr);
		printf("%d ", root->data);
		inorder(root->rptr);
	}
}

int main()
{
	node *root;
	root = newnode(1);
	root->lptr = newnode(2);
	root->rptr = newnode(3);
	root->lptr->lptr = newnode(4);
	root->lptr->rptr = newnode(5);

	printf("In Order Transversal: ");
	inorder(root);

	return 0;
}

File: Tree/Tree_Inorder_Traversal/Tree_Inorder_Traversal.go
package main

import (
	"fmt"
)

type node struct {
	data  int
	left  *node
	right *node
}

func newNode(i int) *node {
	return &node{
		data: i,
	}
}

func (n *node) printInorder() {
	if n == nil {
		return
	}

	n.left.printInorder()
	fmt.Printf("%d ", n.data)
	n.right.printInorder()
}

func main() {
	root := newNode(5)
	root.left = newNode(2)
	root.right = newNode(3)
	root.left.left = newNode(4)
	root.left.right = newNode(5)
	root.printInorder()
}
